cmd/telemetrygen/internal/metrics/worker.go

```diff
@@ -91,13 +91,14 @@ func (w worker) simulateMetrics(res *resource.Resource, exporterFunc func() (sdk
 			ScopeMetrics: []metricdata.ScopeMetrics{{Metrics: metrics}},
 		}
 
-		if err := exporter.Export(context.Background(), &rm); err != nil {
-			w.logger.Fatal("exporter failed", zap.Error(err))
-		}
 		if err := limiter.Wait(context.Background()); err != nil {
 			w.logger.Fatal("limiter wait failed, retry", zap.Error(err))
 		}
 
+		if err := exporter.Export(context.Background(), &rm); err != nil {
```

Comment:
I'm curious about this move: I don't see anything wrong with limiting it before exporting, but did you see a problem with exporting and then waiting?

Reply:
I think this was causing an off-by-one error in the amount of spans I was expecting. In the tests I was adding, I was expecting between 1 and 2 spans/logs/metric points (for the rate/duration of those tests), but kept getting 3. I believe firing off the first data point immediately was causing that extra point. Moving this seemed to fix it and it seemed more in line with the expectation to limit before sending.

Reply:
sounds good to me